cosmic-core/server/src/main/java/com/cloud/api/ApiDispatcher.java

```diff
@@ -120,10 +120,6 @@ public void dispatch(final BaseCmd cmd, final Map<String, String> params, final
             ((BaseCustomIdCmd) cmd).checkUuid();
         }
 
-        try {
-            cmd.execute();
-        } catch (ResourceUnavailableException | InsufficientCapacityException | ResourceAllocationException | NetworkRuleConflictException e) {
-            throw new CloudException("Caught exception while executing command", e);
-        }
+        cmd.execute();
```

Comment:
How do you know that the exception which is being throw is actually the root cause?

With this you can get it for example: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/exception/ExceptionUtils.html#getRootCause-java.lang.Throwable-